packages/button/src/Button.js

```diff
@@ -26,6 +26,8 @@ type Props = {
   disabled?: boolean,
   /** Stretch Button to fill its container */
   fullWidth?: boolean,
+  /** An instance of an Icon component */
+  icon?: MnrlReactNode,
```

Comment:
It may be best to support both `iconStart` and `iconEnd`.  The challenge would then be how to communicate that the primary design intent is to place icons at the start.